tests/unit/commands/init/test_templates.py

```diff
@@ -0,0 +1,59 @@
+import json
+import subprocess
+import click
+
+from mock import mock_open, patch, PropertyMock, MagicMock
+from re import search
+from unittest import TestCase
+
+from samcli.commands.init.init_templates import InitTemplates
+
+
+class TestTemplates(TestCase):
+    @patch("subprocess.check_output")
+    @patch("samcli.commands.init.init_templates.InitTemplates._git_executable")
+    def test_location_from_app_template(self, subprocess_mock, git_exec_mock):
+        it = InitTemplates(True)
+
+        manifest = {
+            "ruby2.5": [
+                {
+                    "directory": "mock-ruby-template",
+                    "displayName": "Hello World Example",
+                    "dependencyManager": "bundler",
+                    "appTemplate": "hello-world",
+                }
+            ]
+        }
+        manifest_json = json.dumps(manifest)
+
+        m = mock_open(read_data=manifest_json)
+        with patch("samcli.cli.global_config.GlobalConfig.config_dir", new_callable=PropertyMock) as mock_cfg:
+            mock_cfg.return_value = "/tmp/test-sam"
+            with patch("samcli.commands.init.init_templates.open", m):
+                location = it.location_from_app_template("ruby2.5", "bundler", "hello-world")
+                assert location == "/tmp/test-sam/aws-sam-cli-app-templates/mock-ruby-template"
```

Comment:
self.assertEqual might be better. in all places where we assert.

Reply:
Done